device_calendar/example/lib/presentation/pages/calendar_event.dart

```diff
@@ -100,23 +99,30 @@ class _CalendarEventPageState extends State<CalendarEventPage> {
         if (_event.recurrenceRule.endDate != null) {
           _recurrenceRuleEndType = RecurrenceRuleEndType.SpecifiedEndDate;
           _recurrenceEndDate = _event.recurrenceRule.endDate;
-          _recurrenceEndTime = TimeOfDay.fromDateTime(_recurrenceEndDate);
         }
-        _daysOfTheWeek =
-            _event.recurrenceRule.daysOfTheWeek ?? List<DayOfTheWeek>();
-        _daysOfTheMonth = _event.recurrenceRule.daysOfTheMonth ?? List<int>();
-        _monthsOfTheYear = _event.recurrenceRule.monthsOfTheYear ?? List<int>();
-        _weeksOfTheYear = _event.recurrenceRule.weeksOfTheYear ?? List<int>();
-        _setPositions = _event.recurrenceRule.setPositions ?? List<int>();
+
+        _isByDayOfMonth = _event.recurrenceRule.setPositions == null ? true : false;
+        _daysOfTheWeek = _event.recurrenceRule.daysOfTheWeek ?? List<DayOfTheWeek>();
+
+        if (_recurrenceFrequency == RecurrenceFrequency.Monthly || _recurrenceFrequency == RecurrenceFrequency.Yearly) {
+          _monthsOfTheYear = _event.recurrenceRule.monthsOfTheYear ?? List<int>();
+
+          if (!_isByDayOfMonth) {
+            _setPositions = _event.recurrenceRule.setPositions ?? List<int>();
+            _selectedDayPos = _daysOfTheWeek?.first ?? DayOfTheWeek.Sunday;
```

Comment:
Setting the beginning of the week as Sunday seems a bit arbitrary. Should we consider the fact that in different countries the first day might differ (i.e. in Poland the first day of week is Monday, whereas in UK it's Sunday)?

Reply:
I couldn't find a way to set Monday as a start of week for Android so I left it as it is before.
I've now changed it to be Monday and not breaking the code by using enum extensions.

Reply:
I'm what's the goal here but I would imagine that this value should be dependant on the current device culture? It might be an overhead for now, so if we won't do that now we should probably leave a TODO in here to look into this later.

It might be that I'm wrong and this can/should/needs to be hard coded to a day so therefore ignore the above

cc. @nickrandolph